### PR TITLE
Convert model files from rst to mdx

### PR DESCRIPTION
**Pull request by LysandreJik**
First pass!

### CODE REVIEW

File: docs/source/model_doc/layoutlmv2.mdx [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+  done by the model. Important to note is that the visual backbone expects BGR channels instead of RGB, as all models
+  in Detectron2 are pre-trained using the BGR format. The `bbox` input are the bounding boxes (i.e. 2D-positions)
+  of the input text tokens. This is identical to [`~transformers.LayoutLMModel`]. These can be obtained using an
+  external OCR engine such as Google's [Tesseract](https://github.com/tesseract-ocr/tesseract) (there's a [Python
+  wrapper](https://pypi.org/project/pytesseract/) available). Each bounding box should be in (x0, y0, x1, y1)
+  format, where (x0, y0) corresponds to the position of the upper left corner in the bounding box, and (x1, y1)
+  represents the position of the lower right corner. Note that one first needs to normalize the bounding boxes to be on
+  a 0-1000 scale. To normalize, you can use the following function:
+
+```
```
Comment (sgugger) [inline docs/source/model_doc/layoutlmv2.mdx:71]:
```suggestion
```python
```

File: docs/source/model_doc/layoutlmv2.mdx [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+         int(1000 * (bbox[2] / width)),
+         int(1000 * (bbox[3] / height)),
+     ]
+```
+
+Here, `width` and `height` correspond to the width and height of the original document in which the token
+occurs (before resizing the image). Those can be obtained using the Python Image Library (PIL) library for example, as
+follows:
+
+```
```
Comment (sgugger) [inline docs/source/model_doc/layoutlmv2.mdx:85]:
```suggestion
```python
```

File: docs/source/model_doc/layoutlmv2.mdx [hunk 3]
```diff
@@ -0,0 +1,287 @@
...
+## Usage: LayoutLMv2Processor
+
+The easiest way to prepare data for the model is to use [`~transformers.LayoutLMv2Processor`], which internally
+combines a feature extractor ([`~transformers.LayoutLMv2FeatureExtractor`]) and a tokenizer
+([`~transformers.LayoutLMv2Tokenizer`] or [`~transformers.LayoutLMv2TokenizerFast`]). The feature extractor
+handles the image modality, while the tokenizer handles the text modality. A processor combines both, which is ideal
+for a multi-modal model like LayoutLMv2. Note that you can still use both separately, if you only want to handle one
+modality.
+
+```
```
Comment (sgugger) [inline docs/source/model_doc/layoutlmv2.mdx:123]:
```suggestion
```python
```

File: docs/source/model_doc/layoutlmv2.mdx [hunk 4]
```diff
@@ -0,0 +1,287 @@
...
+  statistics, which are going to have values when fine-tuning on a custom dataset.
+- If you want to train the model in a distributed environment, make sure to call [`synchronize_batch_norm`] on the
+  model in order to properly synchronize the batch normalization layers of the visual backbone.
+
+In addition, there's LayoutXLM, which is a multilingual version of LayoutLMv2. More information can be found on
+[LayoutXLM's documentation page](layoutxlm).
+
+## Usage: LayoutLMv2Processor
+
+The easiest way to prepare data for the model is to use [`~transformers.LayoutLMv2Processor`], which internally
```
Comment (sgugger) [inline docs/source/model_doc/layoutlmv2.mdx:116]:
```suggestion
The easiest way to prepare data for the model is to use [`LayoutLMv2Processor`], which internally
```

File: docs/source/model_doc/layoutlmv2.mdx [hunk 5]
```diff
@@ -0,0 +1,287 @@
...
+combines a feature extractor ([`~transformers.LayoutLMv2FeatureExtractor`]) and a tokenizer
+([`~transformers.LayoutLMv2Tokenizer`] or [`~transformers.LayoutLMv2TokenizerFast`]). The feature extractor
```
Comment (sgugger) [inline docs/source/model_doc/layoutlmv2.mdx:118]:
```suggestion
combines a feature extractor ([`LayoutLMv2FeatureExtractor`]) and a tokenizer
([`LayoutLMv2Tokenizer`] or [`LayoutLMv2TokenizerFast`]). The feature extractor
```

File: docs/source/model_doc/layoutlmv2.mdx [hunk 6]
```diff
@@ -0,0 +1,287 @@
...
+bounding boxes, as well to resize the image to a given size in order to get the `image` input. The words and
+normalized bounding boxes are then provided to [`~transformers.LayoutLMv2Tokenizer`] or
+[`~transformers.LayoutLMv2TokenizerFast`], which converts them to token-level `input_ids`,
+`attention_mask`, `token_type_ids`, `bbox`. Optionally, one can provide word labels to the processor,
+which are turned into token-level `labels`.
+
+[`~transformers.LayoutLMv2Processor`] uses [PyTesseract](https://pypi.org/project/pytesseract/), a Python
+wrapper around Google's Tesseract OCR engine, under the hood. Note that you can still use your own OCR engine of
+choice, and provide the words and normalized boxes yourself. This requires initializing
+[`~transformers.LayoutLMv2FeatureExtractor`] with `apply_ocr` set to `False`.
```
Comment (sgugger) [inline docs/source/model_doc/layoutlmv2.mdx:143]:
```suggestion
In short, one can provide a document image (and possibly additional data) to [`LayoutLMv2Processor`],
and it will create the inputs expected by the model. Internally, the processor first uses
[`LayoutLMv2FeatureExtractor`] to apply OCR on the image to get a list of words and normalized
bounding boxes, as well to resize the image to a given size in order to get the `image` input. The words and
normalized bounding boxes are then provided to [`LayoutLMv2Tokenizer`] or
[`LayoutLMv2TokenizerFast`], which converts them to token-level `input_ids`,
`attention_mask`, `token_type_ids`, `bbox`. Optionally, one can provide word labels to the processor,
which are turned into token-level `labels`.

[`LayoutLMv2Processor`] uses [PyTesseract](https://pypi.org/project/pytesseract/), a Python
wrapper around Google's Tesseract OCR engine, under the hood. Note that you can still use your own OCR engine of
choice, and provide the words and normalized boxes yourself. This requires initializing
[`LayoutLMv2FeatureExtractor`] with `apply_ocr` set to `False`.
```

**Review by sgugger — APPROVED** [review]
Thanks a lot for converting those models!

File: docs/source/model_doc/layoutlmv2.mdx [hunk 7]
```diff
@@ -0,0 +1,287 @@
...
+
+- The main difference between LayoutLMv1 and LayoutLMv2 is that the latter incorporates visual embeddings during
+  pre-training (while LayoutLMv1 only adds visual embeddings during fine-tuning).
+- LayoutLMv2 adds both a relative 1D attention bias as well as a spatial 2D attention bias to the attention scores in
+  the self-attention layers. Details can be found on page 5 of the [paper](https://arxiv.org/abs/2012.14740).
+- Demo notebooks on how to use the LayoutLMv2 model on RVL-CDIP, FUNSD, DocVQA, CORD can be found [here](https://github.com/NielsRogge/Transformers-Tutorials).
+- LayoutLMv2 uses Facebook AI's [Detectron2](https://github.com/facebookresearch/detectron2/) package for its visual
+  backbone. See [this link](https://detectron2.readthedocs.io/en/latest/tutorials/install.html) for installation
+  instructions.
+- In addition to `input_ids`, [`~transformers.LayoutLMv2Model.forward`] expects 2 additional inputs, namely
```
Comment (sgugger) [inline docs/source/model_doc/layoutlmv2.mdx:57]:
```suggestion
- In addition to `input_ids`, [`~LayoutLMv2Model.forward`] expects 2 additional inputs, namely
```

File: docs/source/model_doc/layoutlmv2.mdx [hunk 8]
```diff
@@ -0,0 +1,287 @@
...
+  `image` and `bbox`. The `image` input corresponds to the original document image in which the text
+  tokens occur. The model expects each document image to be of size 224x224. This means that if you have a batch of
+  document images, `image` should be a tensor of shape (batch_size, 3, 224, 224). This can be either a
+  `torch.Tensor` or a `Detectron2.structures.ImageList`. You don't need to normalize the channels, as this is
+  done by the model. Important to note is that the visual backbone expects BGR channels instead of RGB, as all models
+  in Detectron2 are pre-trained using the BGR format. The `bbox` input are the bounding boxes (i.e. 2D-positions)
+  of the input text tokens. This is identical to [`~transformers.LayoutLMModel`]. These can be obtained using an
```
Comment (sgugger) [inline docs/source/model_doc/layoutlmv2.mdx:64]:
```suggestion
  of the input text tokens. This is identical to [`LayoutLMModel`]. These can be obtained using an
```
